applications: nrf_desktop: turn on BT_ID_AUTO_SWAP_MATCHING_BONDS (#23094)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * The :ref:`CONFIG_DESKTOP_HIDS_SUBSCRIBER_REPORT_MAX <config_desktop_app_options>` Kconfig option to :ref:`nrf_desktop_hids`.
    The option allows you to limit the number of HID input reports that can be simultaneously processed by the module.
    This limits the number of GATT notifications with HID reports in the Bluetooth stack.

>   * The :kconfig:option:`CONFIG_BT_ID_AUTO_SWAP_MATCHING_BONDS` Kconfig option in implied options for nRF Desktop peripherals.

Reviewer: ```suggestion
  * The :kconfig:option:`CONFIG_BT_ID_AUTO_SWAP_MATCHING_BONDS` Kconfig option as an imply for nRF Desktop peripherals.
```

  * The :ref:`CONFIG_DESKTOP_HIDS_SUBSCRIBER_REPORT_MAX <config_desktop_app_options>` Kconfig option to :ref:`nrf_desktop_hids`.
    The option allows you to limit the number of HID input reports that can be simultaneously processed by the module.
    This limits the number of GATT notifications with HID reports in the Bluetooth stack.

>   * The :kconfig:option:`CONFIG_BT_ID_AUTO_SWAP_MATCHING_BONDS` Kconfig option as an imply for Fast Pair nRF Desktop peripherals.
>     The option allows the application to automatically swap matching bonds when the Bluetooth local identity changes.
>     Due to that change and change in the zephyr host smp module, the nRF Desktop application no longer requires the :kconfig:option:`CONFIG_ BT_ID_UNPAIR_MATCHING_BONDS` Kconfig option to be enabled.
>     Removing it enables the possibility for bonding with the same HID host on multiple Bluetooth local identities for all nRF Desktop peripherals.

Reviewer 2: ```suggestion
  * The :kconfig:option:`CONFIG_BT_ID_AUTO_SWAP_MATCHING_BONDS` Kconfig option as an imply to the :kconfig:option:`CONFIG_DESKTOP_BT_PERIPHERAL` Kconfig option.
    As a result, all nRF Desktop Fast Pair configurations have the :kconfig:option:`CONFIG_BT_ID_AUTO_SWAP_MATCHING_BONDS` Kconfig option enabled by default.
    The option allows the application to automatically swap matching bonds when the Bluetooth local identity changes and the Bluetooth Privacy feature (:kconfig:option:`CONFIG_BT_PRIVACY`)  is enabled.
    Build configurations without the Bluetooth Privacy feature support the bond swapping without any dedicated Kconfig option.
    Due to that change and change in the SMP layer of the Zephyr Bluetooth stack, the nRF Desktop application no longer requires the :kconfig:option:`CONFIG_BT_ID_UNPAIR_MATCHING_BONDS` Kconfig option to be enabled.
    Removing it enables the possibility for bonding with the same HID host on multiple Bluetooth local identities for all nRF Desktop peripherals.
```

    To enable the module, set the :ref:`CONFIG_DESKTOP_BLE_ADV_CTRL_ENABLE <config_desktop_app_options>` Kconfig option to ``y``.
    To enable the module to suspend and resume the |ble_adv| when the USB state changes, set the :ref:`CONFIG_DESKTOP_BLE_ADV_CTRL_SUSPEND_ON_USB <config_desktop_app_options>` Kconfig option to ``y``.
    These options are enabled for targets that support the USB High-Speed.
  * The :kconfig:option:`CONFIG_BT_ID_AUTO_SWAP_MATCHING_BONDS` Kconfig option as an imply to the :kconfig:option:`CONFIG_DESKTOP_BT_PERIPHERAL` Kconfig option.
    As a result, all nRF Desktop configurations that are set up for the Bluetooth Peripheral role have the :kconfig:option:`CONFIG_BT_ID_AUTO_SWAP_MATCHING_BONDS` Kconfig option enabled by default.
    The option allows the application to automatically swap matching bonds when the local Bluetooth identity changes.
    Due to that change the nRF Desktop application no longer requires the :kconfig:option:`CONFIG_BT_ID_UNPAIR_MATCHING_BONDS` Kconfig option to be enabled.

>     Removing it enables the possibility for bonding with the same HID host on multiple Bluetooth local identities for all nRF Desktop peripherals.

Reviewer: Please add a comment on PR https://github.com/nrfconnect/sdk-nrf/pull/23631, if this change is for the 3.1.0 release, and remove the entry from this PR.
Also, add the backport label.
